AgContext.yaml

```diff
@@ -0,0 +1,321 @@
+      - tillering
+      - stem elongation
+      - rosette growth
+      - shoot development
+      - development of harvestable vegetative parts
+      - bolting
+      - inflorescence emergence
+      - heading
+      - flowering
+      - development of fruit
+      - ripening or maturity of fruit and seed
+      - senescence
+      - beginning of dormancy
+      - na
+    description: |-
+        BBCG descriptive text.
+        One of several possible strings describing the stage of the crop, chosen from a list of possible terms used by the BBCH.
+        If this AgContext is not in a cropping environment, use value "na".
+
+  bbch_code:
+    type:
+      string
+    enum:
+      - gs01
+      - gs02
+      - gs04
+      - gs05
+      - gs06
+      - gs07
+      - gs08
+      - gs09
+      - gs10
+      - gs11
+      - gs12
+      - gs13
+      - gs14
+      - gs15
+      - gs16
+      - gs17
+      - gs18
+      - gs19
+      - gs20
+      - gs21
+      - gs22
+      - gs23
+      - gs24
+      - gs25
+      - gs26
+      - gs27
+      - gs28
+      - gs29
+      - gs30
+      - gs31
+      - gs32
+      - gs33
+      - gs34
+      - gs35
+      - gs36
+      - gs37
+      - gs38
+      - gs39
+      - gs40
+      - gs41
+      - gs42
+      - gs43
+      - gs44
+      - gs45
+      - gs46
+      - gs47
+      - gs48
+      - gs49
+      - gs50
+      - gs51
+      - gs52
+      - gs53
+      - gs54
+      - gs55
+      - gs56
+      - gs57
+      - gs58
+      - gs59
+      - gs60
+      - gs61
+      - gs62
+      - gs63
+      - gs64
+      - gs65
+      - gs66
+      - gs67
+      - gs68
+      - gs69
+      - gs70
+      - gs71
+      - gs72
+      - gs73
+      - gs74
+      - gs75
+      - gs76
+      - gs77
+      - gs78
+      - gs79
+      - gs80
+      - gs81
+      - gs82
+      - gs83
+      - gs84
+      - gs85
+      - gs86
+      - gs87
+      - gs88
+      - gs89
+      - gs90
+      - gs91
+      - gs92
+      - gs93
+      - gs94
+      - gs95
+      - gs96
+      - gs97
+      - gs98
+      - gs99
+    description: |-
+        BBCG descriptive text.
+        One of several possible strings describing the stage of the crop, chosen from a list of codes used by the BBCH.
+        If this AgContext is not in a cropping environment, use value "na".
+
+  soil_colour:
+    type:
+      string
+    enum:
+      - not_visible
+      - black
+      - dark_brown
+      - brown
+      - red_brown
+      - dark_red
+      - yellow
+      - pale_yellow
+      - white
+      - grey
+    description: |-
+        Soil colour.
+        General description of the soil colour.
+
+  surface_cover:
+    type: string
+    enum:
+      - cereal
+      - oilseed
+      - legume
+      - cotton
+      - black_plastic
+      - white_plastic
+      - woodchips
+      - other
+    description: |-
+      Surface cover type.
+      One of several strings describing the background cover that is behind any plants in the images.
+
+  surface_coverage:
```

Comment:
Would this be better represented as a set of possible number ranges?

Reply:
I'm not sure what you mean by "a set of possible number ranges". Do you mean that these would be "type: number" in JSON schema?